File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution
{
    public:
        bool isIsomorphic(string s, string t)
        {
            int n = s.length();
            string changedS = "";
            unordered_map<char, char> mp1, mp2;
            for (int i = 0; i < n; i++)
            {
                if (!mp1[t[i]] && !mp2[s[i]]) mp1[t[i]] = s[i], mp2[s[i]] = t[i];
                else if (mp2[s[i]] != t[i] || mp1[t[i]] != s[i]) return false;
            }
            return true;;
        }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length()) return false;

        Map<Character, Character>mapST = new HashMap<>();
        Map<Character, Character>mapTS = new HashMap<>();

        for(int i = 0; i < s.length(); i++){
            char charS = s.charAt(i);
            char charT = t.charAt(i);

            if(mapST.containsKey(charS)){
                if(mapST.get(charS) != charT) return false;
            }else{
                mapST.put(charS, charT);
            }

            if(mapTS.containsKey(charT)){
                if(mapTS.get(charT) != charS) return false;
            }else{
                mapTS.put(charT, charS);
            }
        }

        return true;
    }
}